### PR TITLE
need inspect existing connection anyway if new connection is not for the Toolkit

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
Patch #4305 

in the previous PR, the logic will not inspect existing connections if the new connection is of type "AwsBearerTokenConnection".

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt [hunk 1]
```diff
@@ -75,12 +69,7 @@ class AmazonQToolWindowFactory : ToolWindowFactory, DumbAware {
             } ?: false
         } ?: false
 
-        val isQConnected = (isQConnection || isQConnected(project)).also {
-            val old = isConnected.getAndSet(it)
-            if (old == it) {
-                return
-            }
-        }
+        val isQConnected = (isQConnection || isQConnected(project))
```
Comment (manodnyab) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt:72]: what is the difference between the 2?

Comment (Will-ShaoHua) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt:72]: sorry, i will rename these variable names. 

basically the chain is
1. check new connection is Toolkit conn
2. if (1) == false, check existing connection

Comment (Will-ShaoHua) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt:72]: `fun isQConnected` inspects "existing" connection

Comment (Will-ShaoHua) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt:72]: `isQConnection` refers to "is the new connection a Q connection"

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt [hunk 2]
```diff
@@ -75,16 +69,9 @@ class AmazonQToolWindowFactory : ToolWindowFactory, DumbAware {
...
-            val old = isConnected.getAndSet(it)
-            if (old == it) {
-                return
-            }
-        }
-
         // isQConnected alone is not robust and there is race condition (read/update connection states)
-        val component = if (isQConnected) {
-            LOG.debug { "returning Q-chat window; isQConnection=$isQConnection; hasPinnedConnection=$isQConnection" }
+        val component = if ((isNewConnectionForQ || isQConnected(project))) {
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt:73]:
```suggestion
        val component = if (isNewConnectionForQ || isQConnected(project)) {
```

**Review by rli — APPROVED** [review]

**Pull request merged**